packages/app/src/cli/services/build/extension.ts

```diff
@@ -186,7 +186,12 @@ export async function buildFunctionExtension(
       await touchFile(bundlePath)
       await writeFile(bundlePath, base64Contents)
     }
-  } catch (error) {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  } catch (error: any) {
+    const errors = error.errors ?? []
+    // If there is an `errors` array, it's an esbuild error, re-throw it directly.
+    if (errors.length) throw error
```

Comment:
since we never use `errors` again

```suggestion
    // If there is an `errors` array, it's an esbuild error, re-throw it directly.
    if (error.errors?.length) throw error
```